framework/canceledcontext/canceled_context.go

```diff
@@ -0,0 +1,61 @@
+// Package canceledcontext stores and accesses the canceled in
+// context.Context.
+package canceledcontext
+
+import (
+	"context"
+)
+
+// key is an unexported type for keys defined in this package. This prevents
+// collisions with keys defined in other packages.
+type key string
+
+// canceledKey is the key for canceled values in context.Context.
+// Clients use canceledcontext.NewContext and
+// canceledcontext.FromContext instead of using this key directly.
+var canceledKey key = "canceled"
+
+// NewContext returns a new context.Context that carries value v.
+func NewContext(ctx context.Context, v chan struct{}) context.Context {
+	if v == nil {
+		return ctx
+	}
+
+	return context.WithValue(ctx, canceledKey, v)
+}
+
+// FromContext returns the canceled, if any.
```

Comment:
"the canceled context" reads better to me.

Reply:
It is actually the channel that is returned. But thanks for the hint. Done.